acquia_lift.module

```diff
@@ -694,47 +694,58 @@ function acquia_lift_personalize_option_set_save($option_set) {
+  $agent = personalize_agent_load($option_set->agent);
+  if (!acquia_lift_is_targeting_agent($agent)) {
     return;
   }
-
-  // For fields-based option sets, we need to do some more set-up, depending on the
-  // field settings.
-  $field = field_info_field($option_set->data['personalize_fields_field_name']);
-  if (!isset($field['settings']['personalize']) || !$field['settings']['personalize']['enabled']) {
+  // Create a goal for hte option set. If it's a fields-based option set we first
+  // check whether this is the configured behavior.
+  if ($option_set->plugin == 'fields') {
+    $field = field_info_field($option_set->data['personalize_fields_field_name']);
+    if (!isset($field['settings']['personalize']) ||
+      !$field['settings']['personalize']['enabled'] ||
+      !isset($field['settings']['personalize']['create_goal']) ||
+      !$field['settings']['personalize']['create_goal']
+      ) {
+      return;
+    }
+  }
+  // @todo Expose this variable in the UI.
+  elseif (!variable_get('acquia_lift_auto_goal', TRUE)) {
     return;
   }
 
-  if (isset($option_set->is_new)) {
-    if (isset($field['settings']['personalize']['create_goal']) && $field['settings']['personalize']['create_goal']) {
-      $goals = personalize_goal_load_by_conditions(array('agent' => $agent->machine_name));
-      if (empty($goals)) {
-        $js_id = personalize_stringify_osid($option_set->osid);
-        $plugin = 'link';
-        $action_label = t('Clicks @option_set', array('@option_set' => $option_set->label));
-        $action_name = personalize_generate_machine_name($action_label, 'visitor_actions_machine_name_exists', '_');
-        $pages = empty($field['settings']['personalize']['goal_pages']) ? '' : $field['settings']['personalize']['goal_pages'];
-        $action = array(
-          'label' => $action_label,
-          'machine_name' => $action_name,
-          'plugin' => $plugin,
-          'client_side' => 1,
-          'identifier' => '[data-personalize=' . $js_id . ']',
-          'event' => 'click',
-          'pages' => $pages,
-          'data' => array(),
-          'limited_use' => 1
-        );
-        // Allow the plugin to modify the action before saving.
-        if ($class = ctools_plugin_load_class('visitor_actions', 'actionable_element', $plugin, 'handler')) {
-          $action = call_user_func_array(array($class, 'actionPresave'), array($action));
-        }
-        if (visitor_actions_save_action($action)) {
-          personalize_goal_save($option_set->agent, $action_name, 1);
-        }
-      }
+  // Only create a goal if there are no goals already set up for the agent.
+  $goals = personalize_goal_load_by_conditions(array('agent' => $agent->machine_name));
+  if (empty($goals)) {
+    $js_id = personalize_stringify_osid($option_set->osid);
+    $plugin = 'link';
+    $action_label = t('Clicks @option_set', array('@option_set' => $option_set->label));
```

Comment:
The code here is doing that (in fact more nicely cause its the label rather than machine name).  The problem is that we are helpful and automatically suggest the name "Variation set #1", etc. so its likely that many people will have that as a variation set label in multiple campaigns.